src/cart/crt_group.c

```diff
@@ -937,6 +937,15 @@ crt_grp_lc_lookup(struct crt_grp_priv *grp_priv, int ctx_idx,
 	D_ASSERT(uri != NULL || hg_addr != NULL);
 	D_ASSERT(ctx_idx >= 0 && ctx_idx < CRT_SRV_CONTEXT_NUM);
 
+	if (rank == CRT_NO_RANK) {
```

Comment:
you could add a comment that explains why someone would be looking up CRT_NO_RANK...